ui/Screen/IdentityCreation.svelte

```diff
@@ -0,0 +1,175 @@
+<script>
+  import { gql } from "apollo-boost";
+  import { getClient, mutate } from "svelte-apollo";
+  import { pop, push } from "svelte-spa-router";
+  import validatejs from "validate.js";
+
+  import {
+    IMAGE_FILENAME,
+    NAME_MATCH,
+    SINGLE_WORD_MATCH
+  } from "../lib/validationHelpers.js";
+  import { setCurrentIdentity } from "../store/identity.js";
+  import { showNotification } from "../store/notification.js";
+
+  import { Button, Input, Text, Title } from "../DesignSystem/Primitive";
+  import { ModalLayout } from "../DesignSystem/Component";
+
+  let handle,
+    displayName,
+    avatarUrl,
+    validations,
+    beginValidation = false;
+
+  validatejs.options = {
+    fullMessages: false
+  };
+
+  const constraints = {
+    handle: {
+      presence: {
+        message: "You must provide a handle",
+        allowEmpty: false
+      },
+      format: {
+        pattern: SINGLE_WORD_MATCH,
+        message: "Handle should match [a-z0-9][a-z0-9_-]+"
```

Comment:
As the validation messages are coupled with the matchers (i.e. when changing the matcher, you'd have to change the message as well), does it even make sense to extract the regex matchers into a separate file?

Reply:
I think it makes sense because some of the regular expressions can be shared (e.g. the one for identity handles and project names). I also think the components look cleaner without them, but if it'd be simpler to leave them in the `<script>` tags instead of extracting them we can go back to that!